app/Controllers/Donation/AddDonationController.php

```diff
@@ -63,6 +63,10 @@ private function newHttpResponse( AddDonationResponse $responseModel ): Response
 	private function createDonationRequest( Request $request ): AddDonationRequest {
 		$donationRequest = new AddDonationRequest();
 
+		if ( $request->get( 'addressType', '' ) === 'company_with_contact' ) {
+			$request->query->set( 'addressType', AddressType::LEGACY_COMPANY );
+		}
+
```

Comment:
This hack should go into `getSafeDonorType` IMHO, with a comment or maybe even with a restructuring of that code so we know what's going on.